website/server/libs/payments.js

```diff
@@ -259,9 +277,28 @@ api.buyGems = async function buyGems (data) {
     let byUsername = getUserInfo(data.user, ['name']).name;
     let gemAmount = data.gift.gems.amount || 20;
 
-    let message = `\`Hello ${data.gift.member.profile.name}, ${data.user.profile.name} has sent you ${gemAmount} gems!\``;
-    if (data.gift.message) message += ` ${data.gift.message}`;
-    data.user.sendMessage(data.gift.member, message);
+    // generate the message in both languages, so both users can understand it
+    let languages = [data.user.preferences.language, data.gift.member.preferences.language];
+    let senderMsg = shared.i18n.t('giftedGemsFull', {
+      username: data.gift.member.profile.name,
+      sender: byUsername,
+      gemAmount,
+    }, languages[0]);
+    senderMsg = `\`${senderMsg}\``;
+
+    let receiverMsg = shared.i18n.t('giftedGemsFull', {
+      username: data.gift.member.profile.name,
+      sender: byUsername,
+      gemAmount,
+    }, languages[1]);
+    receiverMsg = `\`${receiverMsg}\``;
+
+    if (data.gift.message) {
+      receiverMsg += ` ${data.gift.message}`;
+      senderMsg += ` ${data.gift.message}`;
+    }
+
+    data.user.sendMessage(data.gift.member, { receiverMsg, senderMsg });
```

Comment:
There's the existing test, but it uses the same languages. I could add one akin to the transfer-gems one if you think it is needed.